tests/utils/mock_thread_pool.h

```diff
@@ -42,13 +42,29 @@ class tieredIndexMock {
         size_t k;    // The number of results to return.
         size_t n;    // The number of vectors in the index (might be useful for the mock)
         size_t dim;  // The dimension of the vectors in the index (might be useful for the mock)
```

Comment:
can't these be conculded from the index?

Reply:
I guess... I believe that @GuyAv46 did it like this for convenience